pkg/internal/controller/controller.go

```diff
@@ -228,11 +230,14 @@ func (c *Controller) reconcileHandler(obj interface{}) bool {
 		return true
 	}
 
+	log := c.Log.WithValues("name", req.Name, "namespace", req.Namespace)
+	ctx := logf.SetContext(context.Background(), log)
+
 	// RunInformersAndControllers the syncHandler, passing it the namespace/Name string of the
 	// resource to be synced.
-	if result, err := c.Do.Reconcile(req); err != nil {
+	if result, err := c.Do.Reconcile(ctx, req); err != nil {
 		c.Queue.AddRateLimited(req)
-		c.Log.Error(err, "Reconciler error", "name", req.Name, "namespace", req.Namespace)
+		log.Error(err, "Reconciler error")
```

Comment:
Do we need to potentially worry about a user replacing the logger completely in the context? If so, then it might be good to do a `log = logf.FromContext(ctx)` here (and other codepaths where log is used after this call).

Reply:
I want to remove it if possible, we're logging for users and potentially losing information or polluting the logs

Reply:
Similar to others, I think it would be helpful to have this at a higher verbosity by default so that it could help with debugging type purposes, but by +1 to not outputting it by default